common/src/main/java/com/wynntils/core/chat/ChatModel.java

```diff
@@ -131,31 +130,35 @@ private static void handleMultilineMessage(Component message) {
         if (newLines.isEmpty()) {
             // No new lines has appeared since last registered chat line.
             // We could just have a dialog that disappeared, so we must signal this
-            handleNpcDialog(List.of());
+            handleNpcDialog(List.of(), true);
             return;
         }
 
-        if (newLines.getLast().isEmpty()) {
+        if (newLines.getLast().getString().isEmpty()) {
             // Wynntils add an empty line before the NPC dialog; remove it
             newLines.removeLast();
         }
 
-        LinkedList<String> newChatLines = new LinkedList<>();
+        LinkedList<Component> newChatLines = new LinkedList<>();
         LinkedList<String> dialog = new LinkedList<>();
 
-        if (NPC_FINAL_PATTERN.matcher(newLines.getFirst()).find()) {
+        String firstNewLineCoded = ComponentUtils.getCoded(newLines.getFirst());
+
+        boolean noShiftToContinueDialog = false;
+
+        if (NPC_DIALOGUE_PATTERN.matcher(firstNewLineCoded).matches()) {
+            noShiftToContinueDialog = true;
+            handleNpcDialog(List.of(firstNewLineCoded), false);
+        } else if (NPC_FINAL_PATTERN.matcher(firstNewLineCoded).find()) {
             // This is an NPC dialog screen.
             // First remove the "Press SHIFT to continue" trailer.
             newLines.removeFirst();
-            if (newLines.getFirst().isEmpty()) {
```

Comment:
There used to be an empty line inbetween. Like:
```
xxxxx: ....

[1/3] Key Master: Are you the gatekeeper?

Press SHIFT to continue
```

We needed to remove that, both to trim down the message sent as an event, and to avoid falsely triggering the detection of the "end" (that is, the start, since we're doing this in reverse) of the NPC message.

I don't understand how that is handled if you remove this?

Reply:
We remove all empty lines before this code is reached.